### PR TITLE
Java Hashmap with different value types

### PR DESCRIPTION
**Pull request by sk1418**

### CODE REVIEW

File: core-java-modules/core-java-collections-4/src/test/java/com/baeldung/multipletypesinmap/MultipleTypesInMapLiveTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import java.util.Map;
+
+class MultipleTypesInMapLiveTest {
+
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
+        .withZone(ZoneId.systemDefault());
+
+    @Test
+    void givenThreeTypes_whenUsingRawMap_thenPrintDescription() {
+
```
Comment (KevinGilmore) [inline core-java-modules/core-java-collections-4/src/test/java/com/baeldung/multipletypesinmap/MultipleTypesInMapLiveTest.java:23]: Let's remove the empty line after the opening curly brace

File: core-java-modules/core-java-collections-4/src/test/java/com/baeldung/multipletypesinmap/MultipleTypesInMapLiveTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                    String.format("The value is an instant: %s", FORMATTER.format(theV)));
+            } else {
+                throw new IllegalStateException("Unknown Type found.");
+            }
+        });
+    }
+
+    @Test
+    void givenThreeTypes_whenUsingAnInterface_thenPrintDescription() {
+
```
Comment (KevinGilmore) [inline core-java-modules/core-java-collections-4/src/test/java/com/baeldung/multipletypesinmap/MultipleTypesInMapLiveTest.java:54]: Remove blank line

**Review by KevinGilmore — COMMENTED** [review]
A couple of blank lines to remove

File: core-java-modules/core-java-collections-4/src/test/java/com/baeldung/multipletypesinmap/MultipleTypesInMapLiveTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        Map<String, Object> rawMap = new HashMap<>();
+        rawMap.put("E1 (Integer)", intValue);
+        rawMap.put("E2 (IntArray)", intArray);
+        rawMap.put("E3 (Instant)", instant);
+
+        rawMap.forEach((k, v) -> {
+            if (v instanceof Integer) {
+                Integer theV = (Integer) v;
+                System.out.println(k + " -> " +
+                    String.format("The value is a %s integer: %d.", theV > 0 ? "positive" : "negative", theV));
```
Comment (KevinGilmore) [inline core-java-modules/core-java-collections-4/src/test/java/com/baeldung/multipletypesinmap/MultipleTypesInMapLiveTest.java:37]: For consistency, let's remove the period from the end of the string. The other cases don't include a period.

File: core-java-modules/core-java-collections-4/src/main/java/com/baeldung/collections/mulipletypesinmap/IntegerTypeValue.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    public IntegerTypeValue(Integer value) {
+        this.value = value;
+    }
+
+    @Override
+    public String valueDescription() {
+        if(value == null){
+            return "The value is null.";
+        }
+        return String.format("The value is a %s integer: %d.", value > 0 ? "positive" : "negative", value);
```
Comment (KevinGilmore) [inline core-java-modules/core-java-collections-4/src/main/java/com/baeldung/collections/mulipletypesinmap/IntegerTypeValue.java:15]: Remove the trailing period

Comment (sk1418) [reply core-java-modules/core-java-collections-4/src/main/java/com/baeldung/collections/mulipletypesinmap/IntegerTypeValue.java:15]: fixed.

File: core-java-modules/core-java-collections-4/src/test/java/com/baeldung/multipletypesinmap/MultipleTypesInMapLiveTest.java (hunk 3 above)
Comment (sk1418) [reply core-java-modules/core-java-collections-4/src/test/java/com/baeldung/multipletypesinmap/MultipleTypesInMapLiveTest.java:37]: period removed.

File: core-java-modules/core-java-collections-4/src/test/java/com/baeldung/multipletypesinmap/MultipleTypesInMapLiveTest.java (hunk 2 above)
Comment (sk1418) [reply core-java-modules/core-java-collections-4/src/test/java/com/baeldung/multipletypesinmap/MultipleTypesInMapLiveTest.java:54]: fixed

File: core-java-modules/core-java-collections-4/src/test/java/com/baeldung/multipletypesinmap/MultipleTypesInMapLiveTest.java (hunk 1 above)
Comment (sk1418) [reply core-java-modules/core-java-collections-4/src/test/java/com/baeldung/multipletypesinmap/MultipleTypesInMapLiveTest.java:23]: fixed

**Pull request merged**